client/src/components/molecules/DonationMatchingRequestsTable.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { FunctionComponent, useEffect, useState } from "react";
+import moment from "moment";
+import Table from "react-bootstrap/Table";
+
+import { DonationForm } from "../../data/types/donationForm";
+import Request from "../../data/types/request";
+import ScrollableDropdown from "../atoms/ScrollableDropdown";
+import { TextField } from "../atoms/TextField";
+
+interface Props {
+    requests: Request[];
+    donationForm?: DonationForm;
+    isErroneous: boolean; // indicates whether the selected quantities are erroneous
+    onQuantitySelected: (newQuantity: number, requestId: string) => void;
+}
+
+type RequestQuantities = {
+    quantity: number; // current value selected in dropdown
+    totalRequired: number; // total unmatched requests
+};
+
+const DonationMatchingRequestsTable: FunctionComponent<Props> = (props: Props) => {
+    const headingList = ["Client Name", "Quantity", "Total Required", "Date Requested"];
+
+    const [requests, setRequests] = useState<Request[]>([]);
+    const [reqCurrentlyEditing, setReqCurrentlyEditing] = useState<string>("");
+
+    useEffect(() => {
+        // sort requests by most recent date
+        const sortedRequests: Request[] = props.requests
+            .filter((request) => !request.deleted && !request.fulfilled)
+            .sort((a, b) => {
+                return b!.createdAt!.valueOf() - a!.createdAt!.valueOf();
+            });
+        setRequests(sortedRequests);
+    }, []);
+
+    /**
+     * Calculates the numerical values to display in the table row for a specific request
+     */
+    const valuesToDisplay = (request: Request): RequestQuantities => {
+        const { matchedDonations } = request;
+
+        const totalQuantityMatched = matchedDonations?.reduce((total, contribution) => {
+            return total + contribution.quantity;
+        }, 0) as number;
+
+        const requestTotal = request?.quantity ?? 0;
+        const totalRequired = requestTotal - totalQuantityMatched;
+
+        if (props.donationForm) {
```

Comment:
Is there a scenario where props.donationForm will be null?

Reply:
if `props.DonationForm` is null, that means the user is viewing the list of donation forms, e.g. the "View all Antique robes" page